testsuite/tests/cli/dgw/tls_anchoring.rs

```diff
@@ -0,0 +1,120 @@
+use anyhow::Context as _;
+use rstest::rstest;
+use testsuite::cli::dgw_tokio_cmd;
+use testsuite::dgw_config::{DgwConfig, DgwConfigHandle};
+use tokio::process::Child;
+
+async fn start_gateway() -> anyhow::Result<(DgwConfigHandle, Child)> {
+    let config_handle = DgwConfig::builder()
+        .disable_token_validation(true)
+        .build()
+        .init()
+        .context("init config")?;
+
+    // Start JMUX server that will accept JMUX connections.
+    let process = dgw_tokio_cmd()
+        .env("DGATEWAY_CONFIG_PATH", config_handle.config_dir())
+        .kill_on_drop(true)
+        .stdout(std::process::Stdio::piped())
+        .spawn()
+        .context("failed to start Devolutions Gateway")?;
+
+    // Give the server a moment to start.
+    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+
+    Ok((config_handle, process))
+}
```

Comment:
E2E tests actually spawing a Devolutions Gateway process, and connecting to it the same way a WebBrowser would.